File: tests/fixtures/java/Simple.java
/**
 * FIXTURE: Simple Java class
 * TESTS: Basic method signature extraction
 */

public class Simple {
    private int value;

    public Simple(int value) {
        this.value = value;
    }

    public int add(int a, int b) {
        return a + b;
    }

    public String greet(String name) {
        return "Hello, " + name + "!";
    }

    public static void main(String[] args) {
        Simple calc = new Simple(10);
        System.out.println(calc.add(5, 3));
    }
}

interface Computer {
    int compute(int x);
}

enum Status {
    ACTIVE,
    INACTIVE,
    PENDING
}

File: tests/fixtures/c/simple.c
// FIXTURE: Simple C functions and types
// TESTS: Basic function signature extraction

#include <stdio.h>
#include <stdlib.h>

typedef int Status;

enum Color {
    RED,
    GREEN,
    BLUE
};

struct Point {
    int x;
    int y;
};

int add(int a, int b) {
    return a + b;
}

void greet(const char* name) {
    printf("Hello, %s!\n", name);
}

struct Point make_point(int x, int y) {
    struct Point p;
    p.x = x;
    p.y = y;
    return p;
}
